src/particles/beam/BeamParticleContainer.cpp

```diff
@@ -102,15 +114,6 @@ BeamParticleContainer::ReadParameters ()
         soa.GetIntData()[icomp].setArena(
```

Comment:
```suggestion
        soa.GetIntData(icomp).setArena(
```
This, but for the real data above, is what is causing the segfaults. `.GetRealData()` returns an array of only the compile time components, however the loop over rcomp also includes the 3 spin runtime components from the define(3, 0). Garbage data is written somewhere in memory causing a segfault at the end of the simulation (in my case the capacity of a DeviceVector was set to 6504664838913477633 causing a crash when deallocating itself). The `.GetRealData(rcomp)` version handles runtime components correctly.